communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzerImpl.java

```diff
@@ -80,40 +80,47 @@ public GAVDependencyTree readRelationships(File pomRepoDir, File pomPath)
                 Map<File, ProjectVersionRef> projectVersionRefs = getProjectVersionRefs(tempDir,
                         findAllPomFiles(pomRepoDir));
 
+                ProjectVersionRef rootProject = null;
+
                 // for each pom.xml
                 for (Map.Entry<File, ProjectVersionRef> entry : projectVersionRefs.entrySet()) {
+                    // if pomPath is the same as what we found analysis the root repository
+                    if (pomPath.getCanonicalPath().equals(entry.getKey().getCanonicalPath())) {
+                        rootProject = entry.getValue();
+                        break;
+                    }
+                }
```

Comment:
I believe I used the canonical path so that I could compare the absolute paths of both, and make the comparison more robust.

```java
import java.io.File;

public class Main {
    public static void main (String[] args) throws Exception {
        File file = new File("/home/dcheung/test.py");
        File file2 = new File("../../../../../../../../../../../home/dcheung/test.py");

        System.out.println(file.equals(file2));
        System.out.println(file.getCanonicalPath().equals(file2.getCanonicalPath()));
    }
}
```

This will print:
```
false
true
```